src/DacpacTool/Extensions.cs

```diff
@@ -27,6 +113,29 @@ public static void AddReference(this TSqlModel model, string referencePath, stri
             AddCustomData(dataSchemaModel, customData);
         }
 
+        public static IEnumerable<string> GetReferencedDacPackages(this TSqlModel model)
+        {
+            var result = new List<string>();
+            var dataSchemaModel = GetDataSchemaModel(model);
+
+            var getCustomDataMethod = dataSchemaModel.GetType().GetMethod("GetCustomData", BindingFlags.Public | BindingFlags.Instance, null, new Type[] { typeof(string), typeof(string) }, null);
+            var references = (IEnumerable) getCustomDataMethod.Invoke(dataSchemaModel, new object[] { "Reference", "SqlSchema" });
+
+            MethodInfo getMetadataMethod = null;
+            foreach (var reference in references)
+            {
+                if (getMetadataMethod == null)
+                {
+                    getMetadataMethod = reference.GetType().GetMethod("GetMetadata", BindingFlags.Public | BindingFlags.Instance, null, new Type[] { typeof(string) }, null);
+                }
+
+                var fileName = (string)getMetadataMethod.Invoke(reference, new object[] { "FileName" });
```

Comment:
Add null check?

Reply:
Do you mean on the result of `GetMethod()` or on the result if `Invoke()`?

Reply:
I meant before doing a result.Add()

Reply:
Ah, yes, that makes sense. Added that now.